src/ui/public/navbar/__tests__/navbar.js

```diff
@@ -0,0 +1,117 @@
+const ngMock = require('ngMock');
+const sinon = require('sinon');
+const expect = require('expect.js');
+const angular = require('angular');
+const _ = require('lodash');
+
+require('ui/navbar');
+const navbarExtensionsRegistry = require('ui/registry/navbar_extensions');
+const Registry = require('ui/registry/_registry');
+
+const defaultMarkup = `
+  <navbar name="testing">
+    <div class="button-group" role="toolbar">
+      <button>
+        <i aria-hidden="true" class="fa fa-file-new-o"></i>
+      </button>
+      <button>
+        <i aria-hidden="true" class="fa fa-save"></i>
+      </button>
+      <button>
+        <i aria-hidden="true" class="fa fa-folder-open-o"></i>
+      </button>
+    </div>
+  </navbar>`;
+
+
+describe('navbar directive', function () {
+  let $rootScope;
+  let $compile;
+  let stubRegistry;
+
+  beforeEach(function () {
+    ngMock.module('kibana', function (PrivateProvider) {
+      stubRegistry = new Registry({
+        index: ['name'],
+        group: ['appName'],
+        order: ['order']
+      });
+
+      PrivateProvider.swap(navbarExtensionsRegistry, stubRegistry);
+    });
+
+    ngMock.module('kibana/navbar');
+
+    // Create the scope
+    ngMock.inject(function ($injector) {
+      $rootScope = $injector.get('$rootScope');
+      $compile = $injector.get('$compile');
+    });
+  });
+
+  function init(markup = defaultMarkup) {
+    // Give us a scope
+    const $el = angular.element(markup);
+    $compile($el)($rootScope);
+    $el.scope().$digest();
+    return $el;
+  }
+
+  describe('incorrect use', function () {
```

Comment:
It probably wouldn't hurt to mention some of the other invalid cases:
 - multiple button groups
 - button groups aren't direct children